src/components/blocks/TakeAction.tsx

```diff
@@ -1,98 +1,155 @@
+import type {
+	PageBlocksActions,
+	PageBlocksActionsItems,
+	PageBlocksActionsItemsBodyActionButtonFilter,
+	PageBlocksActionsItemsFilter,
+} from '~tina/__generated__/types'
 
 const components = {
-  ActionButton: (props) => {
-    return (
-      <Button variant='filled' color='#65676B' component={Link} href={`${props.link}`} >{props.actionText}</Button>
-    )
-  }
+	ActionButton: (props: PageBlocksActionsItemsBodyActionButtonFilter) => {
+		return (
+			<Button
+				variant='filled'
+				color='#65676B'
+				component={Link}
+				href={{ pathname: props.link as unknown as string }}
+			>
+				{props.actionText as ReactNode}
+			</Button>
+		)
+	},
 }
 
-export const TakeAction = ({data}) => {
-  return (
-    <div className='border-2 border-gray-400 rounded-lg flex flex-col justify-between items-center gap-3 max-w-lg min-w-full text-black p-4 bg-white'>
-      {data.title &&
-        <h1 className='lg:text-2xl md:text-xl text-2xl font-bold'>{data.title}</h1>
-      }
-      {data.text &&
-        <p className='md:text-md text-sm lg:text-lg'>{data.text}</p>
-      }
-      <TinaMarkdown components={components} content={data.body} />
-    </div>
-  )
+export const TakeAction = ({ data, display }: { data: PageBlocksActionsItems; display: string }) => {
+	return display === 'flex' ? (
+		<div className='md:max-w-[33.333333%] md:w-1/3 p-1 flex flex-grow w-full'>
+			<div className='w-full gap-3 border-2 border-black rounded-lg flex flex-col justify-between items-center text-black p-4 bg-white'>
+				{data?.title && <h1 className='lg:text-2xl md:text-xl text-2xl font-bold'>{data.title}</h1>}
+				{data?.text && <p className='md:text-md text-sm lg:text-lg'>{data.text}</p>}
+				<TinaMarkdown components={components} content={data?.body} />
+			</div>
+		</div>
+	) : (
+		<div className='w-full p-1'>
+			<div className='w-full gap-3 border-2 border-black rounded-lg flex flex-col justify-between items-center text-black p-4 bg-white'>
+				{data?.title && <h1 className='lg:text-2xl md:text-xl text-2xl font-bold'>{data.title}</h1>}
+				{data?.text && <p className='md:text-md text-sm lg:text-lg'>{data.text}</p>}
+				<TinaMarkdown components={components} content={data?.body} />
+			</div>
+		</div>
+	)
 }
-export const TakeActionContainer = ({data}) => {
-  return (
-    <div className='grid md:grid-cols-3 grid-cols-1 p-6 gap-4'>
-      {data.items &&
-        data.items.map((block, i: number) => {
-          return (
-            <TakeAction data={block} key={i} />
-          )
-        })
-      }
-    </div>
-  )
+
+export const TakeActionContainer = ({ data }: { data: PageBlocksActions }) => {
+	return (
+		<div
+			className={
+				data.display === 'grid' ? 'grid md:grid-cols-3 grid-cols-1' : 'flex w-full flex-wrap justify-center'
+			}
+		>
+			{data.items &&
+				data.items.map((block: PageBlocksActionsItems | null, i: number) => {
+					if (block) return <TakeAction data={block} key={i} display={data.display as string} />
```

Comment:
Any way we can address the keys prop here? Shouldn't rely on the index only. Especially since we are able to change the order around for the cards